public/tabs.js

```diff
@@ -23,247 +14,280 @@
 
-      // Update arrays related to tabs widget
-      generateArrays();
+        // Set the tab as selected
+        tab.classList.add('active');
+        tab.parentElement.classList.add('active');
+        tab.setAttribute('aria-selected', 'true');
+
+        // Get the value of aria-controls (which is an ID)
+        var controls = tab.getAttribute('aria-controls');
+
+        // Remove hidden attribute from tab panel to make it visible
+        var tabContent = document.getElementById(controls);
+        if(tabContent === undefined || tabContent === null)
+        {
+          debugger;
+        }
 
-      // Activate the closest tab to the one that was just deleted
-      if (target.index - 1 < 0) {
-        activateTab(tabs[0]);
-      }
-      else {
-        activateTab(tabs[target.index - 1]);
+        tabContent.removeAttribute('hidden');
+        tabContent.setAttribute('tabindex', '0');
+        tabContent.setAttribute('aria-hidden', 'false');
+        tabContent.classList.add('active');
+
+        // Set focus when required
+        if (setFocus) {
+          tab.focus();
+        };
       };
-    };
-  };
 
-  // Deletes a tab and its panel
-  function deleteTab (event) {
-    var target = event.target;
-    var panel = document.getElementById(target.getAttribute('aria-controls'));
+      // Deactivate all tabs and tab panels
+      function deactivateTabs () {
+        for (let t = 0; t < tabs.length; t++) {
+          tabs[t].setAttribute('tabindex', '-1');
+          tabs[t].setAttribute('aria-selected', 'false');
+          tabs[t].classList.remove('active');
+          tabs[t].parentElement.classList.remove('active');
+          tabs[t].removeEventListener('focus', focusEventHandler);
+        }
 
-    target.parentElement.removeChild(target);
-    panel.parentElement.removeChild(panel);
-  };
+        for (let p = 0; p < panels.length; p++) {
+          panels[p].setAttribute('hidden', 'hidden');
+          panels[p].setAttribute('tabindex', '-1');
+          panels[p].setAttribute('aria-hidden', 'true');
+          panels[p].classList.remove('active');
+        }
+      };
 
-  // Determine whether there should be a delay
-  // when user navigates with the arrow keys
-  function determineDelay () {
-    var hasDelay = tablist.hasAttribute('data-delay');
-    var delay = 0;
-
-    if (hasDelay) {
-      var delayValue = tablist.getAttribute('data-delay');
-      if (delayValue) {
-        delay = delayValue;
-      }
-      else {
-        // If no value is specified, default to 300ms
-        delay = 300;
+      // Make a guess
+      function focusFirstTab () {
+        tabs[0].focus();
       };
-    };
 
-    return delay;
-  };
+      // Make a guess
+      function focusLastTab () {
+        tabs[tabs.length - 1].focus();
+      };
 
-  //
-  function focusEventHandler (event) {
-    var target = event.target;
+      // Detect if a tab is deletable
+      function determineDeletable (event) {
```

Comment:
Under what circumstances do we need to determine if a tab is deletable?

Reply:
The function name is misleading because it doesn't just determine that a tab is deletable. It does actually go ahead and delete the tab under certain conditions... Maybe it should be "deleteTabIfDeletable"

Reply:
updated